### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException Android crash

### ISSUE THREAD

**Issue by tombotley** [Adobe-Marketing-Cloud/acp-sdks#528]
I have started seeing the following crash reported in Firebase Test Lab on numerous devices (this one on Pixel API 25) with our React Native app. This was not reported when I tested the app at the end of January and I have made no changes to any SDKs/packages in that time, although I notice that there was an update to sdk-core (1.6.0) in February. I have tried this version as well as rolling back to earlier versions and also adding the latest versions of okhttp and okio to my dependencies but the issue persists.

I also noticed that the release notes for android core 1.6.0 mentions a fix for:

> a crash which was caused by the exception thrown from the Android okhttp library

I wonder if this is the issue I am encountering.

FATAL EXCEPTION: ADBMobileBackgroundThread
```
java.lang.ArrayIndexOutOfBoundsException: length=8192; index=8192
	at com.android.okhttp.okio.Buffer.writeByte(Buffer.java:973)
	at com.android.okhttp.internal.Platform.concatLengthPrefixed(Platform.java:130)
	at com.android.okhttp.internal.Platform.configureTlsExtensions(Platform.java:84)
	at com.android.okhttp.Connection.connectTls(Connection.java:230)
	at com.android.okhttp.Connection.connectSocket(Connection.java:199)
	at com.android.okhttp.Connection.connect(Connection.java:172)
	at com.android.okhttp.Connection.connectAndSetOwner(Connection.java:367)
	at com.android.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:130)
	at com.android.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:329)
	at com.android.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:246)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.execute(HttpURLConnectionImpl.java:457)
	at com.android.okhttp.internal.huc.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:126)
	at com.android.okhttp.internal.huc.DelegatingHttpsURLConnection.connect(DelegatingHttpsURLConnection.java:89)
	at com.android.okhttp.internal.huc.HttpsURLConnectionImpl.connect(HttpsURLConnectionImpl.java)
	at com.adobe.marketing.mobile.HttpConnectionHandler.a()
	at com.adobe.marketing.mobile.AndroidNetworkService.c()
	at com.adobe.marketing.mobile.AndroidNetworkService.b()
	at com.adobe.marketing.mobile.AnalyticsHitsDatabase.l()
	at com.adobe.marketing.mobile.AnalyticsHitsDatabase.i()
	at com.adobe.marketing.mobile.AnalyticsHitsDatabase.a()
	at com.adobe.marketing.mobile.HitQueue$1.run()
	at java.lang.Thread.run(Thread.java:761)
```

app\build.gradle
```
    implementation 'com.adobe.marketing.mobile:analytics:1.+'
    implementation 'com.adobe.marketing.mobile:userprofile:1.+'
    implementation 'com.adobe.marketing.mobile:sdk-core:1.+'
```

package.json
```
  "dependencies": {
    "@adobe/react-native-acpanalytics": "^1.3.0",
    "@adobe/react-native-acpcore": "^1.5.0",
    ...
  }
```

**Comment by shalehaha** [comment]
@tombotley 
We just released another version today (https://github.com/Adobe-Marketing-Cloud/acp-sdks/blob/master/android/core-1.7.0.aar), which includes a fix for this crash. 

Also, I'm curious is there a consistent way you can reproduce this crash? Does it only happen on Adobe' SDK, or it also happens from app's own net requests? How often are you seeing this crash?

Jiabin

**Comment by tombotley** [comment]
Thanks @shalehaha, I have updated to 1.7.0 and the crash is no longer occurring. I only observed this crash when putting my latest builds through Firebase Test Lab, not in production (last release was back in December) or during my own physical or emulator device testing. Of 165 device/android API combinations in Firebase Test Lab around 20 would experience this crash with each test run (some of which I have listed below). It seemed specific to devices with an android API level of 25-28 (not reported on API level 19, 21-24 or 29-30). It appeared completely random as to when and where the crash would occur during the test and it wasn't happening with any network requests other than Adobe SDK.

HTC U11 plus, API Level 26 (Android 8)
HTC U11, API Level 26 (Android 8)
Huawei Honor Play COR-L29, API Level 27 (Android 8.1)
MI 5s, API Level 26 (Android 8)
MI 8 Pro, API Level 28 (Android 9)
moto e5 play, API Level 26 (Android 8)
moto e5 play, API Level 27 (Android 8.1)
Moto Z3 Play, API Level 27 (Android 8.1)
motorola one, API Level 28 (Android 9)
Nokia 7 plus, API Level 28 (Android 9)
Nokia 8 Sirocco, API Level 26 (Android 8)
Nokia 9, API Level 28 (Android 9)
ONEPLUS A3003, API Level 26 (Android 8)
Pixel 2, API Level 28 (Android 9)
Pixel 3, API Level 28 (Android 9)
Pixel, API Level 25 (Android 7.1)
Pixel, API Level 28 (Android 9)
Sharp Aquos R SHV39, API Level 28 (Android 9)
Sharp SH-03K, API Level 28 (Android 9)
vivo 1805, API Level 27 (Android 8.1)
Xperia XZ Premium G8142, API Level 26 (Android 8)
Xperia XZ1 G8342, API Level 26 (Android 8)
Xperia XZ2 H8324, API Level 26 (Android 8)
